src/_pytest/runner.py

```diff
@@ -264,7 +266,10 @@ def pytest_make_collect_report(collector) -> CollectReport:
             skip_exceptions.append(unittest.SkipTest)  # type: ignore
         if call.excinfo.errisinstance(tuple(skip_exceptions)):
             outcome = "skipped"
-            r = collector._repr_failure_py(call.excinfo, "line").reprcrash
+            r_ = collector._repr_failure_py(call.excinfo, "line")
```

Comment:
This looks good to me, but looking at the type of `_repr_failure_py`, I wonder:

1. Why do `_repr_failure_py` and `repr_failure` both exist? `repr_failure` just forwards directly `_repr_failure_py` so why not just inline it?
2. Is it correct that it can only take `Failed` and `FixtureLookupError` and not other exception types?
3. Is it possible to add overloads to it so this assert is not needed?
4. Should its `style` argument be typed with `_TracebackStyle`?

Reply:
1. it is used by other subclasses etc - it is rather tangled (I have some WIP to fix this, also for 4.)
2. IIRC we/I've limited this intentionally for what has been observed, but from the code it is clear that this is not correct: https://github.com/pytest-dev/pytest/blob/719e0e354bd9e13c5a9ecc3f1ac9839d3e7a8704/src/_pytest/nodes.py#L288-L292
3. overload might be a good idea, but in general it would be better to not return a string there.  Also it is based on `excinfo.value.pytrace` - so would need separate types for this I assume.
4. yes, I have that in the WIP to clean up handling of `--fulltrace`